src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -636,7 +636,7 @@ impl<'a, 'tcx> DocFolder for LinkCollector<'a, 'tcx> {
                 }
 
                 match disambiguator.map(Disambiguator::ns) {
-                    Some(ns @ ValueNS) => {
+                    Some(ns @ ValueNS | ns @ TypeNS) => {
```

Comment:
```suggestion
                    Some(ns @ (ValueNS | TypeNS)) => {
```

Reply:
I actually tried to do this earlier with `ns @ ValueNS | TypeNS` but it gave an error, I didn't realize it would work if I added parentheses 😆